modules/web/src/app/node-data/basic/provider/vmware-cloud-director/component.ts

```diff
@@ -309,6 +386,27 @@ export class VMwareCloudDirectorBasicNodeDataComponent
     this._cdr.detectChanges();
   }
 
+  private _setDefaultComputePolicy(computePolicies: VMwareCloudDirectorComputePolicy[]): void {
+    this.computePolicies = computePolicies;
+    this.selectedPlacementPolicy = this._nodeDataService.nodeData.spec.cloud.vmwareclouddirector?.placementPolicy;
+    this.selectedSizingPolicy = this._nodeDataService.nodeData.spec.cloud.vmwareclouddirector?.sizingPolicy;
+
+    if (
+      this.selectedPlacementPolicy &&
+      !computePolicies?.find(policy => policy.name === this.selectedPlacementPolicy)
+    ) {
+      this.selectedPlacementPolicy = '';
+    }
+
+    if (this.selectedSizingPolicy && !computePolicies?.find(policy => policy.name === this.selectedSizingPolicy)) {
+      this.selectedSizingPolicy = '';
+    }
+
+    this.placementPolicyLabel = computePolicies?.length ? PlacementPolicyState.Ready : PlacementPolicyState.Empty;
+    this.sizingPolicyLabel = computePolicies?.length ? SizingPolicyState.Ready : SizingPolicyState.Empty;
```

Comment:
Just in case all compute policies are placement or sizing type.
```suggestion
    this.placementPolicyLabel = this.placementPolicies.length ? PlacementPolicyState.Ready : PlacementPolicyState.Empty;
    this.sizingPolicyLabel = this.sizingPolicies.length ? SizingPolicyState.Ready : SizingPolicyState.Empty;
```